detection_rules/mixins.py

```diff
@@ -112,6 +116,55 @@ def to_dict(self, strip_none_values=True) -> dict:
         return serialized
 
 
+class LockDataclassMixin:
```

Comment:
This class has about four duplicate methods from `MarshmallowDataclassMixin` in this module. Can we abstract these out?

Reply:
No because the mixin depends on the hierarchy of the inheritence - if we inherit one level deeper, it tries to generate schemas off of the secondary object, breaking the model. I think it is good practice in this case to keep the mixins flat

Reply:
Makes sense!